src/t8_forest/t8_forest_cxx.cxx

```diff
@@ -1242,7 +1243,23 @@ t8_forest_element_point_batch_inside (t8_forest_t forest, t8_locidx_t ltreeid, c
   const t8_geometry_c *geometry = t8_cmesh_get_tree_geometry (cmesh, cgtreeid);
   T8_ASSERT (t8_geom_is_linear (geometry));
 #endif
+  if (geom_is_axis_aligned) {
+    double v_min[3];
+    double v_max[3];
 
+    /*Geometry is fully described by v_min and v_max*/
+    t8_forest_element_coordinate (forest, ltreeid, element, 0, v_min);
+    t8_forest_element_coordinate (forest, ltreeid, element, 1, v_max);
+
+    for (int ipoint = 0; ipoint < num_points; ipoint++) {
+      /* A point is inside if it is inbetween the x/y/z-coordinates of v_min and v_max */
+      is_inside[ipoint]
+        = v_min[0] <= points[ipoint * 3] && points[ipoint * 3] <= v_max[0] &&         /* check x-coordinate*/
+          v_min[1] <= points[ipoint * 3 + 1] && points[ipoint * 3 + 1] <= v_max[1] && /* check y-coordinate*/
+          v_min[2] <= points[ipoint * 3 + 2] && points[ipoint * 3 + 2] <= v_max[2];   /* check z-coordinate*/
+    }
```

Comment:
There is no check, that vertex 0 has all the min coords and vertex 1 has all the max coords. But maybe this is a good requirement, since it allows us to use these assumptions for code optimization

Reply:
This check is now performed during cmesh_commit in debug-modus